src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.cs

```diff
@@ -70,43 +64,39 @@ internal JsonObject(JsonElement element, JsonNodeOptions? options = null) : this
             _jsonElement = element;
         }
 
-        internal override JsonNode InternalDeepClone()
+        /// <summary>
+        /// Gets or creates the underlying dictionary containing the properties of the object.
+        /// </summary>
+        internal JsonPropertyDictionary<JsonNode?> Dictionary => _dictionary is { } dictionary ? dictionary : InitializeDictionary();
+
+        internal override JsonNode DeepCloneCore()
         {
-            if (_jsonElement.HasValue)
+            GetUnderlyingRepresentation(out JsonPropertyDictionary<JsonNode?>? dictionary, out JsonElement? jsonElement);
```

Comment:
Refactored to avoid potential races with the intializer method that clears out the `_jsonElement` field.